src/Spanner/Session/SessionPoolInterface.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Copyright 2016 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Spanner\Session;
+
+use Google\Cloud\Spanner\Database;
+
+/**
+ * Describes a session pool.
+ */
+interface SessionPoolInterface
+{
+    const CONTEXT_READ = 'r';
+    const CONTEXT_READWRITE = 'rw';
+    const SESSION_EXPIRATION_SECONDS = 3600;
+
+    /**
+     * Acquire a session from the pool.
+     *
+     * @param string $context The type of session to fetch. May be either `r`
+     *        (READ) or `rw` (READ/WRITE). **Defaults to** `r`.
+     * @return Session
+     * @throws \RunTimeException
+     */
+    public function acquire($context);
+
+    /**
+     * Release a session back to the pool.
+     *
+     * @param Session $session
+     */
+    public function release(Session $session);
+
+    /**
+     * Clear the session pool.
```

Comment:
That will be up to the implementation, currently it simply clears the sessions from the cache and does not delete. I was concerned about the case where we have a high number of sessions in the pool, it didn't seem ideal to have to run x number of synchronous calls to delete every item when we could wait an hour for them to expire. What are your thoughts on the best approach?